StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1122UnitTests.cs

```diff
@@ -0,0 +1,159 @@
+    /// <see cref="SA1122CodeFixProvider"/>.
+    /// </summary>
+    [TestClass]
+    public class SA1122UnitTests : CodeFixVerifier
+    {
+        protected static readonly DiagnosticResult[] EmptyDiagnosticResults = { };
+
+        public string DiagnosticId { get; } = SA1122UseStringEmptyForEmptyStrings.DiagnosticId;
+
+        [TestMethod]
+        public async Task TestEmptySource()
+        {
+            var testCode = @"";
+            await VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+        private async Task TestEmptyStringLiteral(bool useVerbatimLiteral)
+        {
+            var testCode = @"public class Foo
+{{
+    public void Bar()
+    {{
+        string test = {0}"""";
+    }}
+}}";
+
+            DiagnosticResult[] expected;
+
+            expected =
+                new[]
+                {
+                    new DiagnosticResult
+                    {
+                        Id = DiagnosticId,
+                        Message = "Use string.Empty for empty strings",
+                        Severity = DiagnosticSeverity.Warning,
+                        Locations =
+                            new[]
+                            {
+                                new DiagnosticResultLocation("Test0.cs", 5, 23)
+                            }
+                    }
+                };
+
+            await VerifyCSharpDiagnosticAsync(string.Format(testCode, useVerbatimLiteral ? "@" : ""), expected, CancellationToken.None);
+        }
+        public async Task TestWhitespaceStringLiteral(bool useVerbatimLiteral)
+        {
+            var testCode = @"public class Foo
+{{
+    public void Bar()
+    {{
+        string test = {0}""  "";
+    }}
+}}";
+            await VerifyCSharpDiagnosticAsync(string.Format(testCode, useVerbatimLiteral ? "@" : ""), EmptyDiagnosticResults, CancellationToken.None);
+        }
+        public async Task TestNullInMethod()
+        {
+            var testCode = @"public class Foo
+{{
+    public void Bar()
+    {{
+        string test = null;
+    }}
+}}";
+            await VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        private async Task TestSimpleCodeFix(bool useVerbatimLiteral)
+        {
+            string oldSource = @"public class Foo
+{{
+    public void Bar()
+    {{
+        string test = {0}"""";
+    }}
+}}";
+            string newSource = @"public class Foo
+{
+    public void Bar()
+    {
+        string test = string.Empty;
+    }
+}";
+
+            await VerifyCSharpFixAsync(string.Format(oldSource, useVerbatimLiteral ? "@" : ""), newSource);
+        }
+
+        [TestMethod]
+        public async Task TestCodeFixMultipleNodes()
+        {
+            // Tests if the code fix works if the SourceSpan of the diagnostic has more then one SynatxNode associated with it
```

Comment:
I wouldnt have thought about this too if i would not have tested this with a small project I have with a lot of weird syntax in it ;)